### ISSUE TITLE
Incompatible with Gradle Configuration Cache

### ISSUE THREAD

**Issue by PaulWoitaschek** [getsentry/sentry-android-gradle-plugin#39]
The sentry gradle plugin ins incompatible with the gradle configuration cache.
https://docs.gradle.org/current/userguide/configuration_cache.html

The stacktrace is as follows:
```
81 problems were found storing the configuration cache, 3 of which seem unique.
- Task `:app:addSentryProguardSettingsForPlayRelease` of type `io.sentry.android.gradle.SentryProguardConfigTask`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfiguration', a subtype of 'org.gradle.api.artifacts.Configuration', as these are not supported with the configuration cache.
  See https://docs.gradle.org/6.8.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:app:addSentryProguardSettingsForPlayRelease` of type `io.sentry.android.gradle.SentryProguardConfigTask`: cannot serialize object of type 'org.gradle.api.internal.project.DefaultProject', a subtype of 'org.gradle.api.Project', as these are not supported with the configuration cache.
  See https://docs.gradle.org/6.8.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
- Task `:app:addSentryProguardSettingsForPlayRelease` of type `io.sentry.android.gradle.SentryProguardConfigTask`: cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfigurationContainer', a subtype of 'org.gradle.api.artifacts.ConfigurationContainer', as these are not supported with the configuration cache.
  See https://docs.gradle.org/6.8.2/userguide/configuration_cache.html#config_cache:requirements:disallowed_types
See the complete report at file:///builds/yazio/android/app/build/reports/configuration-cache/een4gb9jxnugr2xi1hjizyb6g/configuration-cache-report.html
FAILURE: Build failed with an exception.
* What went wrong:
Configuration cache state could not be cached: field 'applicationVariant' from type 'io.sentry.android.gradle.SentryProguardConfigTask': error writing value of type 'com.android.build.gradle.internal.api.ApplicationVariantImpl'
> Configuration cache state could not be cached: field 'unitTestVariant' from type 'com.android.build.gradle.internal.api.ApplicationVariantImpl': error writing value of type 'com.android.build.gradle.internal.api.UnitTestVariantImpl'
```

I'm using the following versions:
* AGP 7.0.0-alpha05
* Gradle 6.8.2
* sentry gradle plugin 1.7.36

**Comment by marandaneto** [comment]
@PaulWoitaschek thanks for reporting, yes, we are currently planning a full rewrite of this plugin #2

**Comment by marandaneto** [comment]
https://docs.gradle.org/current/userguide/configuration_cache.html

> The configuration cache is an incubating feature, and the details described here may change.

let's wait a bit more before working on it as its still incubating

**Comment by marandaneto** [comment]
should be fixed by #166 and #165, can you test it out with the latest version @PaulWoitaschek ? thanks

**Comment by marandaneto** [comment]
you may need to treat errors as warnings though, due to an AGP issue:

`org.gradle.unsafe.configuration-cache-problems=warn`

> 1 problem was found storing the configuration cache.
> - Task `:app:collectReleaseDependencies` of type `com.android.build.gradle.internal.tasks.PerModuleReportDependenciesTask`: invocation of 'Task.project' at execution time is unsupported.
>   See https://docs.gradle.org/7.2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution

but that's not Sentry Android Gradle Plugin's fault
